fastfield_codecs/src/compact_space/mod.rs

```diff
@@ -171,10 +171,10 @@ pub struct IPCodecParams {
 
 impl CompactSpaceCompressor {
     /// Taking the vals as Vec may cost a lot of memory. It is used to sort the vals.
-    pub fn train_from(column: &impl Column<u128>) -> Self {
+    pub fn train_from(iter: impl Iterator<Item = u128>, num_vals: u64) -> Self {
```

Comment:
Would ExactSizeIterator be usable here?

Reply:
in principle yes, but the api would be much harder to use